source/AI.cpp

```diff
@@ -938,6 +938,12 @@ void AI::Step(Command &activeCommands)
 				continue;
 			}
 		}
+		// Update any orders NPCs may have been given by their associated mission.
+		else if(it->IsSpecial() && !it->IsYours() && it->HasTravelDirective())
+		{
+			const Planet *destination = it->AllStopoversVisited() ? it->GetDestinationPlanet() : nullptr;
+			IssueNPCOrders(*it, it->GetDestinationSystem(), it->GetStopovers(), destination);
```

Comment:
These parameters seem like they ought to be retrieved inside `IssueNPCOrders`, not here.
It's already being given a reference to the `Ship` and the rest of these just come from that object anyway.
It would make that method much easier to read to know where those variables are coming from, rather than having to guess or come up here and check.